src/searchbar.cpp

```diff
@@ -183,20 +182,17 @@ void SearchBarLineEdit::updateCompletion()
 
 void SearchBarLineEdit::openCompletion(const QModelIndex &index)
 {
-    if (m_urlList.size() != 0) {
-        openCompletion(index.data().toString(), index.row());
-    }
-}
-
-void SearchBarLineEdit::openCompletion(const QString& text, int index)
-{
-    QUrl url;
-    if (this->text().compare(text, Qt::CaseInsensitive) == 0) {
-        url = m_urlList.at(index);
-    } else {
-        url = m_urlList.last();
+    if (m_suggestionModel.rowCount() != 0) 
+    {
+        QUrl url;
+        auto editText = index.data().toString();
+        if (this->text().compare(editText, Qt::CaseInsensitive) == 0) {
+            url = index.data(Qt::UserRole).toUrl();
+        } else {
+            url = m_suggestionModel.lastIndex().data(Qt::UserRole).toUrl();
+        }
+        QTimer::singleShot(0, [=](){KiwixApp::instance()->openUrl(url, false);});
     }
-    QTimer::singleShot(0, [=](){KiwixApp::instance()->openUrl(url, false);});
 }
```

Comment:
I think it's better to remove the parameter from `SearchBarLineEdit::openCompletion()` and rename it to `openDefaultCompletion()`

Reply:
I don't believe we can remove the parameter, this function opens the completion based on the user-selected index in `focusInEvent`.

Reply:
I see. I only payed attention to its usage in `SearchBarLineEdit::updateCompletion()` where it is called directly with an argument of `m_suggestionModel.index(0)` and missed its usage in `SearchBarLineEdit::focusInEvent()`. But then the code logic is a little confusing for the scenario when a suggestion is selected from the list - it must be clarified that upon selection of an item from the completer popup the content of the search box is updated with the text of the selected item.

Besides the name of the variable `editText` seems to be misleading - that semantics better fits the expression `this->text()` rather than `index.data().toString()` which is better described by `selectedSuggestionText` or `selectedItemText`.

Reply:
Done